superset-frontend/src/explore/components/DataTableControl.tsx

```diff
@@ -0,0 +1,98 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useMemo } from 'react';
+import { styled, t } from '@superset-ui/core';
+import { FormControl } from 'react-bootstrap';
+import debounce from 'lodash/debounce';
+import Button from 'src/components/Button';
+import {
+  applyFormattingToTabularData,
+  prepareCopyToClipboardTabularData,
+} from 'src/utils/common';
+import CopyToClipboard from 'src/components/CopyToClipboard';
+import RowCountLabel from './RowCountLabel';
+
+export const CopyButton = styled(Button)`
+  font-size: ${({ theme }) => theme.typography.sizes.s}px;
+
+  // needed to override button's first-of-type margin: 0
+  && {
+    margin: 0 ${({ theme }) => theme.gridUnit * 2}px;
+  }
+
+  i {
+    padding: 0 ${({ theme }) => theme.gridUnit}px;
+  }
+`;
+
+export const CopyToClipboardButton = ({ data }: { data: object }) => (
+  <CopyToClipboard
+    text={data ? prepareCopyToClipboardTabularData(data) : ''}
```

Comment:
if data is always an object, then this check shouldn't be needed

Reply:
It actually can be undefined, I changed `data: object` to `data?: Record<string, any>`